core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -311,9 +311,11 @@ abstract class AbstractIndex(@volatile private var _file: File, val baseOffset:
   }
 
   protected def safeForceUnmap(): Unit = {
-    try forceUnmap()
-    catch {
-      case t: Throwable => error(s"Error unmapping index $file", t)
+    if (mmap != null) {
```

Comment:
#7387 added null check to ```forceUnmap``` rather than ```safeForceUnmap```. Which one makes sense? For another, ```forceUnmap``` is used by ```safeForceUnmap``` only. We can merge them so above question gets resolved.

Reply:
This one and hence why I did it this way.

Reply:
To clarify, we do use `forceUnmap` directly in a test. It's difficult to test methods that swallow all exceptions and hence it's useful to have `forceUnmap` from that perspective.